### ISSUE TITLE
Error with typo3 when getting redirected to website

### ISSUE THREAD

**Issue by JeremyMancel** [Cidaas/Cidaas-for-Typo3#5]
When getting redirected to website after login in successfully to the IDP,  an error is thrown : `UnknownClassException: Class Authentication does not exist.`

Some bits of the stacktrace : 
```
"TYPO3.CMS.Frontend.ContentObject.Exception.ProductionExceptionHandler": Oops, an error occurred! Code: 2021122216180350590c8a- 
UnknownClassException: Class Authentication does not exist. Reflection failed. Class : Authentication , in file /.../typo3/sysext/extbase/Classes/Reflection/ReflectionService.php:101 - 
{
  "exception":"ReflectionException: Class Authentication does not exist in /.../typo3/sysext/extbase/Classes/Reflection/ClassSchema.php:127
  Stack trace:
    #0 /.../typo3/sysext/extbase/Classes/Reflection/ClassSchema.php(127): ReflectionClass->__construct()
...
}
```

Any ideas ?
Thanks.

**Comment by sawi** [comment]
@JeremyMancel thanks for your ticket. We will check on the issue and update you as soon as we have any findings.

**Comment by JeremyMancel** [comment]
Hello @sawi, forgot to mention this, I'm using your plugin on Typo3 v11.5.4

**Comment by BenoitNorrin** [comment]
Hello. I have this problem too with TYPO3 v11.5.10.

@JeremyMancel Did you find a workaround ?
Thanks

**Comment by BenoitNorrin** [comment]
Problem seems coming from plugin registration : https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/10.0/Deprecation-87550-UseControllerClassesWhenRegisteringPluginsmodules.html

```php
# ext_localconf.php

\TYPO3\CMS\Extbase\Utility\ExtensionUtility::configurePlugin(
    'Widas.' . $_EXTKEY,
    'Pi1',
    [ \Widas\Cidaas\Controller\AuthenticationController::class => 'connect' ],
    [ \Widas\Cidaas\Controller\AuthenticationController::class => 'connect' ]
);
```

Now I got another issue, so let's fix it step by step.

**Comment by JeremyMancel** [comment]
> Hello. I have this problem too with TYPO3 v11.5.10.
> 
> @JeremyMancel Did you find a workaround ? Thanks

No, not really. We decided to go back to v10 since a lot of the plugin we use are still not available on v11.
But if you find any solution please let me know as it may be useful later on !

**Comment by BenoitNorrin** [comment]
Here several changes I made to make it works under TYPO3 v11. More tests are needed.
Those changes probably break compatibility with old TYPO3 versions ... 

https://github.com/Cidaas/Cidaas-for-Typo3/compare/master...BenoitNorrin:master
